src/widget/OktaSignIn.js

```diff
@@ -23,7 +24,9 @@ var OktaSignIn = (function () {
             authClient: authClient,
             globalSuccessFn: (res) => {
               successFn && successFn(res); // call success function if provided
-              resolve(res);
+              if (res.status === Enums.SUCCESS) {
```

Comment:
what if the status is not success.. but the pipeline still ended without any error?  for eg in case of `redirects` ?

Reply:
in this case the promise will not resolve (this is current behavior for redirects)

Reply:
what's the difference between calling `globalSuccessFn` and than again check for `res.status` ? what was missing here ?

Reply:
The global success function can be called multiple times (checking res.status), but the promise can only resolve one time, on success. Apps that need to handle `res.status` that is not `SUCCESS` will still need to pass a successFn.